scheduler/pkg/store/memory.go

```diff
@@ -326,67 +326,47 @@ func (m *MemoryStore) updateLoadedModelsImpl(
 		return m.updateLoadedModelsImpl(modelKey, model.Latest().GetVersion(), serverKey, replicas)
 	}
 
-	// Update model that need to be placed on a replica to request loading
+	//  resevere memory for existing replicas that are not already loading or loaded
 	updatedReplicas := make(map[int]bool)
-	updated := false
 	for _, replica := range replicas {
-		existingState := modelVersion.replicas[replica.GetReplicaIdx()]
+		existingState := modelVersion.replicas[replica.replicaIdx]
 		if !existingState.State.AlreadyLoadingOrLoaded() {
 			logger.Debugf(
 				"Setting model %s version %d on server %s replica %d to LoadRequested",
-				modelKey, modelVersion.version, serverKey, replica.GetReplicaIdx(),
-			)
-			modelVersion.SetReplicaState(replica.GetReplicaIdx(), LoadRequested, "")
-			m.updateReservedMemory(LoadRequested, serverKey, replica.GetReplicaIdx(), modelVersion.GetRequiredMemory())
-			updated = true
-		} else {
-			logger.Debugf(
-				"model %s on server %s replica %d already loaded",
-				modelKey, serverKey, replica.GetReplicaIdx(),
+				modelKey, modelVersion.version, serverKey, replica.replicaIdx,
 			)
+			modelVersion.SetReplicaState(replica.replicaIdx, LoadRequested, "")
+			m.updateReservedMemory(LoadRequested, serverKey, replica.replicaIdx, modelVersion.GetRequiredMemory())
 		}
-		updatedReplicas[replica.GetReplicaIdx()] = true
+		updatedReplicas[replica.replicaIdx] = true
```

Comment:
nit: 
I think that we use `updatedReplicas` in a subtle way:
step 1: essentially we do mark the server replicas that we are going to load the model replica  onto
step 2: for any existing replicas `ReplicaState()`  that are not part of `updatedReplicas` we unload the model replica from them.

It is then less legible that also when unloading the model replica we also set `updatedReplicas` (instead of marking a flag `updated`), I think we should document this behaviour in the code so that it is clearer.

Reply:
Yeah, I can make it more explicit.

Reply:
`updatedReplicas` was essentially a map of the ids of all the `replicas` and it's populated as part of the loop to update replicas that are not already loading, so I've moved it to the first pass of the replicas to validate them, which I think makes the whole method read much more clearly.